dev-packages/e2e-tests/test-applications/vue-3/tests/performance.test.ts

```diff
@@ -0,0 +1,102 @@
+import { expect, test } from '@playwright/test';
+import { waitForTransaction } from '../event-proxy-server';
+
+test('sends a pageload transaction with a parameterized URL', async ({ page }) => {
+  const transactionPromise = waitForTransaction('vue-3', async transactionEvent => {
+    return !!transactionEvent?.transaction && transactionEvent.contexts?.trace?.op === 'pageload';
+  });
+
+  await page.goto(`/users/456`);
+
+  const rootSpan = await transactionPromise;
+
+  expect(rootSpan).toMatchObject({
+    contexts: {
+      trace: {
+        data: {
+          params: {
+            id: '456',
+          },
+          'sentry.source': 'route',
+          'sentry.origin': 'auto.pageload.vue',
+          'sentry.op': 'pageload',
+        },
+        op: 'pageload',
+        origin: 'auto.pageload.vue',
+      },
+    },
+    transaction: '/users/:id',
+    transaction_info: {
+      source: 'route',
+    },
+  });
+});
+
+test('sends a navigation transaction with a parameterized URL', async ({ page }) => {
+  const pageloadTxnPromise = waitForTransaction('vue-3', async transactionEvent => {
+    return !!transactionEvent?.transaction && transactionEvent.contexts?.trace?.op === 'pageload';
+  });
+
+  const navigationTxnPromise = waitForTransaction('vue-3', async transactionEvent => {
+    return !!transactionEvent?.transaction && transactionEvent.contexts?.trace?.op === 'navigation';
+  });
+
+  await page.goto(`/`);
+  await pageloadTxnPromise;
+
+  await page.waitForTimeout(5000);
+
+  const [_, navigationTxn] = await Promise.all([page.locator('#navLink').click(), navigationTxnPromise]);
+
+  expect(navigationTxn).toMatchObject({
+    contexts: {
+      trace: {
+        data: {
+          params: {
+            id: '123',
+          },
+          'sentry.source': 'route',
+          'sentry.origin': 'auto.navigation.vue',
+          'sentry.op': 'navigation',
+        },
+        op: 'navigation',
+        origin: 'auto.navigation.vue',
+      },
+    },
+    transaction: '/users/:id',
+    transaction_info: {
+      // So this is weird. The source is set to custom although the route doesn't have a name.
+      // This also only happens during a navigation. A pageload will set the source as 'route'.
+      // TODO: Figure out what's going on here.
+      source: 'custom',
```

Comment:
Something to maybe tackle in or after #10477: For some reason we set source `'custom'` during a navigation. 
Not sure why this is happening but IMO it should be `'route'`, also when navigating. My guess is it's because we set `custom` when we take the route name over the actual route "path". Probably this is an edge case, we just didn't test before. 

cc @myda if you find an easy fix for this. Otherwise we follow up on this later.